### PR TITLE
[mlir][OpenMP][NFC] Removing unnecessary builders for wsloop

### PR DESCRIPTION
**Pull request by shraiysh**
This patch removes the builders for omp.wsloop operation that aren't specifically needed anywhere. We can add them later if the need arises.

This patch also removes the private/firstprivate/lastprivate operands from `omp.wsloop` to bring it in sync with llvm/main.

Differential Revision (for builders): https://reviews.llvm.org/D120533
Differential Revision (for private/firstprivate/lastprivate operands): https://reviews.llvm.org/D120029

### CODE REVIEW

File: flang/lib/Lower/OpenMP.cpp
Comment (schweitzpgi) [inline flang/lib/Lower/OpenMP.cpp:536]:
```suggestion
  llvm::SmallVector<Value, 4> lowerBound, upperBound, step, linearVars,
```

File: mlir/lib/Dialect/OpenMP/IR/OpenMPDialect.cpp
Comment (schweitzpgi) [inline mlir/lib/Dialect/OpenMP/IR/OpenMPDialect.cpp:1006]:
```suggestion
  llvm::SmallVector<ClauseType> clauses = {
```

File: flang/lib/Lower/OpenMP.cpp
Comment (schweitzpgi) [reply flang/lib/Lower/OpenMP.cpp:536]: nit: In lowering, the namespace is specified to be consistent and avoid class name conflicts and/or errors.

**Comment by shraiysh** [comment]
Ping!

**Comment by jeanPerier** [comment]
The patch looks good to me, but given mostly touches OpenMP parts I am not very familiar with, please wait for @kiranchandramohan approval (I see he approved the related LLVM patch).

**Review by schweitzpgi — COMMENTED** [review]
LGTM. Ditto on getting Kiran to review.

**Review by kiranchandramohan — APPROVED** [review]
LGTM.

**Comment by shraiysh** [comment]
Thanks for the review. I have rebased the patch. Once it passes CI, it can be merged.

**Pull request merged**
